flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/PulsarSourceBuilder.java

```diff
@@ -177,9 +171,11 @@ public PulsarSourceBuilder<OUT> setSubscriptionName(String subscriptionName) {
      * @return this PulsarSourceBuilder.
      * @see <a href="https://pulsar.apache.org/docs/en/concepts-messaging/#subscriptions">Pulsar
      *     Subscriptions</a>
+     * @deprecated We don't support setting the subscription type now.
      */
+    @Deprecated
     public PulsarSourceBuilder<OUT> setSubscriptionType(SubscriptionType subscriptionType) {
```

Comment:
If it's a dummy method, shall we remove these methods? Otherwise, although users' code doesn't fail on compile, the result is actually unexpected.

Reply:
The flink-python still use this method. We should deprecate this method first. Then we can remove this method in next release.

Reply:
@syhily Please refer to the correct code snippet. https://issues.apache.org/jira/browse/FLINK-python is a broken link.